source/crypto/signing.cpp

```diff
@@ -39,116 +40,110 @@ namespace Signing {
+      }
+      k2.absorb(keyFragment, fragmentOffset, ByteHashLength);
+    }
+    k2.finalSqueeze(digests, i * ByteHashLength);
+    k2.reset();
+  }
+  return digests;
+}
+
+std::vector<uint8_t>
+address(const std::vector<uint8_t>& digests) {
+  Kerl                 k;
+  std::vector<uint8_t> addressBytes(ByteHashLength);
+
+  k.absorb(digests);
+  k.finalSqueeze(addressBytes);
+  return addressBytes;
+}
+
+std::vector<uint8_t>
+digest(const std::vector<int8_t>&  normalizedBundleFragment,
+       const std::vector<uint8_t>& signatureFragment) {
   Kerl k1;
   Kerl k2;
 
   for (unsigned int i = 0; i < FragmentLength; i++) {
-    Types::Trits buffer(&signatureFragment[i * TritHashLength],
-                        &signatureFragment[(i + 1) * TritHashLength]);
-    for (unsigned int j = normalizedBundleFragment[i] + NormalizedTryteValueUpperBound; j-- > 0;) {
+    std::vector<uint8_t> buffer(&signatureFragment[i * ByteHashLength],
+                                &signatureFragment[(i + 1) * ByteHashLength]);
+    for (unsigned int j = normalizedBundleFragment[i] + NormalizedTryteUpperBound; j > 0; --j) {
       k2.reset();
       k2.absorb(buffer);
-      k2.squeeze(buffer);
+      k2.finalSqueeze(buffer);
     }
     k1.absorb(buffer);
   }
-  Types::Trits buffer(TritHashLength);
-  k1.squeeze(buffer);
+  std::vector<uint8_t> buffer(ByteHashLength);
+  k1.finalSqueeze(buffer);
   return buffer;
 }
 
-Types::Trits
-digests(const Types::Trits& key) {
-  Kerl         k;
-  auto         numKeys = key.size() / (TritHashLength * FragmentLength);
-  Types::Trits digests;
-
-  for (unsigned int i = 0; i < numKeys; ++i) {
-    Types::Trits keyFragment(key.begin() + (i * TritHashLength * FragmentLength),
-                             key.begin() + ((i + 1) * TritHashLength * FragmentLength));
-    for (unsigned int j = 0; j < FragmentLength; ++j) {
-      Types::Trits buffer(keyFragment.begin() + (j * TritHashLength),
-                          keyFragment.begin() + ((j + 1) * TritHashLength));
-      for (unsigned int l = 0; l < FragmentLength - 1; ++l) {
-        k.reset();
-        k.absorb(buffer);
-        k.squeeze(buffer);
-      }
-      // TODO:1.2.0(optimization) optimize
-      for (unsigned int l = 0; l < 243; ++l) {
-        keyFragment[j * 243 + l] = buffer[l];
-      }
-    }
-
-    k.reset();
-    k.absorb(keyFragment);
-    Types::Trits buffer(TritHashLength);
-    k.squeeze(buffer);
-    digests.insert(std::end(digests), std::begin(buffer), std::end(buffer));
-  }
-  return digests;
-}
-
-Types::Trits
-address(const Types::Trits& digests) {
-  Kerl         k;
-  Types::Trits addressTrits(TritHashLength);
-  k.absorb(digests);
-  k.squeeze(addressTrits);
-  return addressTrits;
-}
-
+// TODO(Optimize)
 Types::Trits
 signatureFragment(const std::vector<int8_t>& normalizedBundleFragment,
                   const Types::Trits&        keyFragment) {
   Kerl         k;
-  Types::Trits signatureFragment;
+  Types::Trits signatureFragment{};
```

Comment:
not sure initializer is useful here, but you can leave it if you want

Reply:
Weird test purposes, will be removed thanks.